src/dotnetinteropt/backend.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (Python) 2024 SCEWO AG
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+import os
+import shutil
+import subprocess  # noqa: S404
+import time
+
+import toml
+
+__all__ = ["install_nugets"]
+
+_DOTNET_LIB_CSPROJ = """<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net{}</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+{}
+  </ItemGroup>
+
+</Project>
+"""
+
+_DOTNET_PACKAGE_REF = """<PackageReference Include="{}" Version="{}" />"""
+
+
+@dataclass
+class _DotnetConfig:
+    package_name: str
+    version: str
+    outpath: str
+    dependencies: list[list[str]]
+    release: bool
+
+
+def _check_dependencies() -> None:
+
+    if shutil.which("dotnet") is None:
+        raise RuntimeError("Please install dotnet first and add it to your PATH "
+                           "(https://dotnet.microsoft.com/en-us/download)")
+
+
+def _get_dotnet_config() -> _DotnetConfig:
+    res = subprocess.run(["dotnet", "--version"], check=True, capture_output=True)  # noqa: S603 S607
+    version = res.stdout.decode("utf-8").split(".")[0].strip()
```

Comment:
As discussed, in the future eventually we might want to make this parametrizable but I wouldn't make this change now 👍